packages/block-library/src/query-loop/index.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { loop } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+import edit from './edit';
+import save from './save';
+
+const { name } = metadata;
+export { metadata, name };
+
+export const settings = {
+	title: __( 'Query Loop' ),
+	icon: loop,
+	parent: [ 'core/query' ],
```

Comment:
What If I want to add a "group" between  the query and the loop, it seems like a valid use-case for styling/layout. 
It seems like there might be  a need for something like "parent" but works across levels.

Reply:
Wouldn't the group be added outside the "query" in that case?

Reply:
Sometimes  you need it to be inside depending on the layout and how you want to place the other "query blocks" (pagination). 

Say you want a background around loop, and a different background around pagination and potentially add some text (paragraph block) between them as a title for "pagination"

I don't think it's a blocker for this PR but I do think we'd need advance use-cases like that.

Reply:
I guess I've been thinking of query and loop as a single unit, with pagination sitting outside and inheriting context globally (since it needs to be reflected in the url structure). Related: https://github.com/WordPress/gutenberg/pull/22199#discussion_r422056316

Reply:
I plan on removing `parent` once we implement a root query context derived from the URL and template.

At that point, you can have the loop and pagination blocks at any level of the document. The Query block is just for setting your own custom queries.